### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by jnogle**
Added initial access rules (Kyle--need to test--will open issue), added api endpoint to register user and corresponding function triggered to add this new user to database, and api endpoint to delete a user and corresponding function triggered to remove this user from the database

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -1,8 +1,215 @@
...
+                    "message": "Phone number must only include numbers, Phone number must be 10 digits",
+                    "name": "ValidatorError",
+                    "path": "phone",
+                    "type": "format"
+                }
+            }
+        }
+        return res.status(400).json(json_res);
+    }
+	if(typeof password === 'undefined' || password.length === 0) {
```
Comment (fedorareis) [inline functions/index.js]: Does the auth section not do its own password validation?

File: functions/index.js [hunk 2]
```diff
@@ -1,8 +1,215 @@
...
+                    "message": "First and Last name required.",
+                    "name": "ValidatorError",
+                    "path": "name",
+                    "type": "required"
+                }
+            }
+        }
+            return res.status(400).json(json_res);
+    }
+    if(password.length < 8 || password.length > 72) {
```
Comment (fedorareis) [inline functions/index.js]: This should be moved to be right next to the other Password validation.

File: functions/index.js [hunk 3]
```diff
@@ -1,8 +1,215 @@
...
+		console.error("Error adding new user to database: ", error);
+	});
+});
+
+/*******************************************************************
+* Author: Jacob Nogle ------------------------------------- 1/2018 *
+* Function to remove a user from the database                      *
+* Triggered by user deletion                                       *
+********************************************************************/
+exports.removeUser = functions.auth.user().onDelete(event => {
```
Comment (fedorareis) [inline functions/index.js:205]: I'm a little confused as to the purpose of this function and how it differs from the delete function above.

**Review by fedorareis — CHANGES_REQUESTED** [review]
Address my other comments and fix the indentation on the index file. The indentation seems to be all over the place.

**Comment by jnogle** [comment]
- Fixed indentation
- Moved password validation statements next to one another
- To answer your first question, no, the auth sdk methods to create users do not do any of their own password validation
- To the second question, firebase stores users in a user pool separate from any database. Our users will obviously be created in a user pool for authentication purposes, but also written to the database so we can store additional information that the user pool does not account for (you can only store username, password, phone, photo there). One of the delete functions I've written (the REST endpoint) deletes the user from the user pool, and the second one removes that user from the database (this is automatically triggered by the user being removed from the pool). You'll notice the same pattern for creating users/adding them to the database.

**Review by fedorareis — APPROVED** [review]

**Pull request merged**